pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/ExpressionConditionalScript.groovy

```diff
@@ -0,0 +1,51 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+
+package org.jenkinsci.plugins.pipeline.modeldefinition.when.impl
+
+import org.jenkinsci.plugins.pipeline.modeldefinition.when.DeclarativeStageConditional
+import org.jenkinsci.plugins.pipeline.modeldefinition.when.DeclarativeStageConditionalScript
+import org.jenkinsci.plugins.workflow.cps.CpsScript
+
+
+class ExpressionConditionalScript extends DeclarativeStageConditionalScript<ExpressionConditional> {
+    public ExpressionConditionalScript(CpsScript s, ExpressionConditional c) {
+        super(s, c)
+    }
+
+    @Override
+    public boolean evaluate() {
+        Object retVal = null
+        Closure c = describable.block?.closure
+
+        if (c != null) {
+            c.delegate = script
+            c.resolveStrategy = Closure.DELEGATE_FIRST
+            retVal = c.call()
+        }
+
+        return ExpressionConditional.booleanFromReturn(retVal)
```

Comment:
What I'm saying is change this line to

```groovy
    if (retVal) {
        return true
    } else {
        return false
    }
```

Then we would say `true` if what was returned was for example a non empty string or a int of value `0` etc. So it behaves like it did in previous version(s).